### ISSUE TITLE
Visual studio code intellisense is very slow

### ISSUE THREAD

**Issue by shanalikhan** [Microsoft/vscode#35210]
Hi, 

Visual Studio Code intellisense is very slow. Sometime it dont provide intellisense after waiting by pressing CTRL + SPACE and most of time it keeps showing loading.

Following is my tsconfig.json file
```
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "outDir": "../out-tsc/app",
    "baseUrl": "./",
    "module": "es2015",
    "types": []
  },
  "exclude": [
    "test.ts",
    "**/*.spec.ts",
    "node_modules",
    "typings/main.d.ts",
    "typings/main"
  ]
}

```

**Comment by adelgado** [comment]
I have the same issue. I noticed it happens a lot when I'm typing `require(`. So much that I've taken to type `rquire('module')` and then add the `e` later. I've only noticed this for a few days, now.

**Comment by mjbvz** [comment]
Sorry, lost track of this issue.

Can you please try installing [VSCode 1.18 insiders](https://code.visualstudio.com/insiders/) which includes TypeScript 2.6 and seeing if the performance issues are still present

**Comment by rstraub** [comment]
I experienced similar issues, they seem to be resolved when using the 1.18 insiders build.

**Comment by dhmorrison** [comment]
I have the 1.18 insiders build and I have the same issue with "Loading" dialog taking an unacceptably long time when working with typescript.  I think it may be related to "omnisharp" being pathetically slow.  

In the case of omnisharp, I adjusted the "projectLoadTimeout" to 6000 so it can load without timing out.  That gives me C# support that is pretty slow in terms of intellisense ui tips, etc.  When omnisharp is running, the typescript intellisense support is basically useless, as it is always "loading...".  Once ts intellisense finally works, I may be onto something else, having given up, and then my cursor is hijacked to near the location where I previously attempted to use ts intellisense.  This is not a great user experience.  My solution is not to load omnisharp unless I'm doing C# work.  This means that I need to segment my work by language ts vs c#.  This is not a great solution.  I would love a fix for this so I don't need to jump through these silly hoops.

Thanks,

David

**Comment by mjbvz** [comment]
@dhmorrison Please try filing this issue against the omnisharp extension

Closing this issue as possibly resolved by VS Code 1.18. @shanalikhan Let me know if you still run into this with VS code insiders

**Comment by CreamFraiche** [comment]
so you are just cool with your app running slow for no reason on multiple languages?

**Comment by dhmorrison** [comment]
Well now, I wouldn't say that.  Nope.  It's not the app per se.  It's vs
code.  Perhaps the vs code team is ok with it.  I'm not sure.

**Comment by mjbvz** [comment]
Yes please open new issues if you are seeing problems. `Visual studio code intellisense is very slow` is super generic and could have any number of different root causing, including things beyond our control such as a misbehaving extension

**Comment by CreamFraiche** [comment]
I have an extensionless install, and "Visual studio code intellisense is very slow". I am coding in typescript, and it was slow before and after the most recent update. Seems pretty relevant to this particular github issue

**Comment by mjbvz** [comment]
As I said, please file a new issue. We will mark it as a duplicate if needed
